debezium-server-pulsar/src/main/java/io/debezium/server/pulsar/PulsarChangeConsumer.java

```diff
@@ -152,4 +175,52 @@ public void handleBatch(List<ChangeEvent<Object, Object>> records, RecordCommitt
         }
         committer.markBatchFinished();
     }
+
+    @SuppressWarnings("unchecked")
+    private void handleBatchAsync(List<ChangeEvent<Object, Object>> records, RecordCommitter<ChangeEvent<Object, Object>> committer) throws InterruptedException {
+        List<CompletableFuture<MessageId>> futures = new ArrayList<>();
+
+        for (ChangeEvent<Object, Object> record : records) {
+            final TypedMessageBuilder message = buildMessage(record);
+
+            final ChangeEvent<Object, Object> currentRecord = record;
+
+            CompletableFuture<MessageId> future = message
+                    .sendAsync()
+                    .thenAccept(messageId -> {
+                        LOGGER.trace("Sent message with id: {}", messageId);
+                        try {
+                            committer.markProcessed(currentRecord);
```

Comment:
I am not fully convinced on this one. We cannot expect that the recordds will be delivered in the order, can we? If not then the last committed offset might not be the latest one form the batch,